### PR TITLE
pen 1370: ci: Add github action testing on pr linting and tests...

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
Lint and test via github action on pr and updated pr

## Jira Ticket
- [PEN-1370](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=545&projectKey=PEN&modal=detail&selectedIssue=PEN-1370&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria

- Github actions! 
- linting!
- testing!
- CI! 

engine theme sdk WPMedia/engine-theme-sdk 

npm t runs test https://github.com/WPMedia/engine-theme-sdk/blob/stable/package.json#L19 

the tests should also be seen in the prs against these branches as well as a check

if all the tests do not pass, the pr should not be able to be merged 

You may need to use secrets in github ui to run the github actions for npm i

add coverage check 

Create a jest config with a base check for coverage (see blocks for example jest.config.js ) 

You will need to create another command `npm run test:coverage` that is similar to https://github.com/WPMedia/fusion-news-theme-blocks/blob/stable/package.json#L22

If the coverage does not pass, the pr should not be able to mergable 

Figure out what the base coverage is for engine theme sdk, then make that coverage just below it 

```

------------------------------|----------|----------|----------|----------|-------------------|
File                          |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------------|----------|----------|----------|----------|-------------------|
All files                     |    53.82 |    68.96 |    60.78 |    54.34 |                   |
 src                          |        0 |        0 |        0 |        0 |                   |
  index.ts                    |        0 |        0 |        0 |        0 |                   |
 src/components               |        0 |        0 |        0 |        0 |                   |
  index.ts                    |        0 |        0 |        0 |        0 |                   |
 src/components/ArcLogo       |      100 |      100 |      100 |      100 |                   |
  index.tsx                   |      100 |      100 |      100 |      100 |                   |
 src/components/EventEmitter  |      100 |      100 |      100 |      100 |                   |
  index.tsx                   |      100 |      100 |      100 |      100 |                   |
 src/components/Gallery       |    81.05 |       82 |    81.48 |    80.85 |                   |
  index.tsx                   |    77.03 |    81.25 |    85.71 |    77.03 |... 95,198,321,322 |
  setInterval.ts              |    90.91 |      100 |    66.67 |       90 |                12 |
  styled.ts                   |      100 |      100 |      100 |      100 |                   |
 src/components/Image         |    90.91 |    74.19 |      100 |    90.91 |                   |
  SourceHandler.tsx           |      100 |      100 |      100 |      100 |                   |
  index.tsx                   |    86.67 |    61.11 |      100 |    86.67 |            90,107 |
  thumbor-image-url.ts        |    90.91 |    88.89 |      100 |    90.91 |                30 |
 src/components/ImageMetadata |      100 |      100 |      100 |      100 |                   |
  index.tsx                   |      100 |      100 |      100 |      100 |                   |
 src/components/Lightbox      |    25.92 |    18.15 |    26.02 |    26.33 |                   |
  constant.ts                 |      100 |      100 |      100 |      100 |                   |
  index.ts                    |        0 |        0 |        0 |        0 |                   |
  lightbox-react.tsx          |    20.62 |    16.25 |    23.28 |    20.86 |... 1716,1836,1912 |
  styled.ts                   |      100 |      100 |      100 |      100 |                   |
  util.ts                     |       50 |       50 |    66.67 |    53.33 |... 12,37,41,42,47 |
 src/components/MetaData      |    98.53 |    95.16 |      100 |    98.52 |                   |
  index.tsx                   |      100 |    97.53 |      100 |      100 |    17,131,153,157 |
  promoImageHelper.ts         |    90.91 |    79.17 |      100 |    90.91 |             12,19 |
 src/components/Video         |      100 |       80 |      100 |      100 |                   |
  index.tsx                   |      100 |       80 |      100 |      100 |                30 |
 src/components/icons         |     96.2 |     98.7 |     98.7 |     96.2 |                   |
  ArrowDownIcon.tsx           |      100 |      100 |      100 |      100 |                   |
  ArrowLeftIcon.tsx           |      100 |      100 |      100 |      100 |                   |
  ArrowRightIcon.tsx          |      100 |      100 |      100 |      100 |                   |
  ArrowUpIcon.tsx             |      100 |      100 |      100 |      100 |                   |
  BackwardIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  BellIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  CalendarIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  CameraIcon.tsx              |      100 |      100 |      100 |      100 |                   |
  ChartIcon.tsx               |      100 |      100 |      100 |      100 |                   |
  CheckIcon.tsx               |      100 |      100 |      100 |      100 |                   |
  ChevronDownIcon.tsx         |      100 |      100 |      100 |      100 |                   |
  ChevronLeftIcon.tsx         |      100 |      100 |      100 |      100 |                   |
  ChevronRightIcon.tsx        |      100 |      100 |      100 |      100 |                   |
  ChevronUpIcon.tsx           |      100 |      100 |      100 |      100 |                   |
  CloseIcon.tsx               |      100 |      100 |      100 |      100 |                   |
  ClosedCaptioningIcon.tsx    |      100 |      100 |      100 |      100 |                   |
  CogIcon.tsx                 |      100 |      100 |      100 |      100 |                   |
  CommentsIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  DeleteIcon.tsx              |      100 |      100 |      100 |      100 |                   |
  DownloadIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  EditIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  EllipsisHorizontalIcon.tsx  |      100 |      100 |      100 |      100 |                   |
  EllipsisVerticalIcon.tsx    |      100 |      100 |      100 |      100 |                   |
  EnvelopeIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  EnvelopeOpenIcon.tsx        |      100 |      100 |      100 |      100 |                   |
  ExternalLinkIcon.tsx        |      100 |      100 |      100 |      100 |                   |
  FacebookAltIcon.tsx         |      100 |      100 |      100 |      100 |                   |
  FacebookIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  ForwardIcon.tsx             |      100 |      100 |      100 |      100 |                   |
  FullscreenIcon.tsx          |      100 |      100 |      100 |      100 |                   |
  GiftIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  GridIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  GripVerticalIcon.tsx        |      100 |      100 |      100 |      100 |                   |
  HamburgerMenuIcon.tsx       |      100 |      100 |      100 |      100 |                   |
  HeadphonesIcon.tsx          |      100 |      100 |      100 |      100 |                   |
  HomeIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  Icon.tsx                    |        0 |        0 |        0 |        0 |  8,11,18,36,43,51 |
  InstagramIcon.tsx           |      100 |      100 |      100 |      100 |                   |
  LinkIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  LinkedInAltIcon.tsx         |      100 |      100 |      100 |      100 |                   |
  LinkedInIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  LockIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  MediumIcon.tsx              |      100 |      100 |      100 |      100 |                   |
  NextIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  PageIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  PauseCircleIcon.tsx         |      100 |      100 |      100 |      100 |                   |
  PauseIcon.tsx               |      100 |      100 |      100 |      100 |                   |
  PhoneIcon.tsx               |      100 |      100 |      100 |      100 |                   |
  PinterestAltIcon.tsx        |      100 |      100 |      100 |      100 |                   |
  PinterestIcon.tsx           |      100 |      100 |      100 |      100 |                   |
  PlayCircleIcon.tsx          |      100 |      100 |      100 |      100 |                   |
  PlayIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  PlusIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  PreviousIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  PrintIcon.tsx               |      100 |      100 |      100 |      100 |                   |
  RedditIcon.tsx              |      100 |      100 |      100 |      100 |                   |
  RssIcon.tsx                 |      100 |      100 |      100 |      100 |                   |
  SearchIcon.tsx              |      100 |      100 |      100 |      100 |                   |
  ShareIcon.tsx               |      100 |      100 |      100 |      100 |                   |
  ShoppingCartIcon.tsx        |      100 |      100 |      100 |      100 |                   |
  SnapchatIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  SoundCloudIcon.tsx          |      100 |      100 |      100 |      100 |                   |
  SoundOffIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  SoundOnIcon.tsx             |      100 |      100 |      100 |      100 |                   |
  StarHalfIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  StarIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  TumblrIcon.tsx              |      100 |      100 |      100 |      100 |                   |
  TwitchIcon.tsx              |      100 |      100 |      100 |      100 |                   |
  TwitterAltIcon.tsx          |      100 |      100 |      100 |      100 |                   |
  TwitterIcon.tsx             |      100 |      100 |      100 |      100 |                   |
  UnlockIcon.tsx              |      100 |      100 |      100 |      100 |                   |
  UserIcon.tsx                |      100 |      100 |      100 |      100 |                   |
  WhatsAppIcon.tsx            |      100 |      100 |      100 |      100 |                   |
  YoutubeAltIcon.tsx          |      100 |      100 |      100 |      100 |                   |
  YoutubeIcon.tsx             |      100 |      100 |      100 |      100 |                   |
  ZoomInIcon.tsx              |      100 |      100 |      100 |      100 |                   |
  ZoomOutIcon.tsx             |      100 |      100 |      100 |      100 |                   |
  index.ts                    |        0 |        0 |        0 |        0 |                   |
  types.ts                    |        0 |        0 |        0 |        0 |                   |
 src/utils                    |        0 |        0 |        0 |        0 |                   |
  Version.tsx                 |        0 |      100 |        0 |        0 |               4,5 |
  index.ts                    |        0 |        0 |        0 |        0 |                   |
  localizeDate.ts             |        0 |        0 |        0 |        0 |... 13,14,16,19,20 |
  localizeDateTime.ts         |        0 |        0 |        0 |        0 |... 13,14,16,19,20 |
------------------------------|----------|----------|----------|---------

### CODE REVIEW

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Cache Node Modules
+        id: cache
+        uses: actions/cache@v2
+        with:
+          path: node_modules
+          key: node-modules-${{ hashFiles('package-lock.json') }}
+
+      - name: Install dependencies if lockfile changed
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: npm install
```
Comment (brentmiller1973) [inline .github/workflows/pr-test.yml]: While not as important as in the fusion-news-theme-blocks repo, still think using `npm ci` is a nice safeguard to ensuring tests run accurately.

**Review by brentmiller1973 — APPROVED** [review]
Let another comment on using npm ci over npm install, but not as vital as it is in fusion-news-theme-blocks, so I leave it up to you

File: .github/workflows/pr-test.yml (hunk 1 above)
Comment (JackHowa) [reply .github/workflows/pr-test.yml]: huh interesting

Comment (JackHowa) [reply .github/workflows/pr-test.yml]: will do

Comment (JackHowa) [reply .github/workflows/pr-test.yml]: updated ✅

**Pull request merged**